Solutions/Corvus.SpecFlow.Extensions/Corvus/SpecFlow/Extensions/FunctionsBindings.cs

```diff
@@ -14,22 +15,57 @@ namespace Corvus.SpecFlow.Extensions
     [Binding]
     public class FunctionsBindings
     {
-        private readonly FunctionsController functionsController;
-        private readonly FeatureContext featureContext;
         private readonly ScenarioContext scenarioContext;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="FunctionsBindings"/> class.
         /// </summary>
-        /// <param name="featureContext">The current feature context.</param>
         /// <param name="scenarioContext">The current scenario context.</param>
-        public FunctionsBindings(FeatureContext featureContext, ScenarioContext scenarioContext)
+        public FunctionsBindings(ScenarioContext scenarioContext)
         {
-            this.functionsController = new FunctionsController();
-            this.featureContext = featureContext;
             this.scenarioContext = scenarioContext;
         }
 
+        /// <summary>
+        /// Retrieves the current functions controller from the supplied context.
+        /// </summary>
+        /// <param name="context">The SpecFlow context to retrieve from.</param>
+        /// <returns>The FunctionsController.</returns>
+        /// <remarks>
+        /// If the controller hasn't already been added to the context, this method will create
+        /// and add a new instance.
+        /// </remarks>
+        public static FunctionsController GetFunctionsController(SpecFlowContext context)
+        {
+            if (!context.TryGetValue(out FunctionsController controller))
+            {
+                controller = new FunctionsController();
+                context.Set(controller);
+            }
+
+            return controller;
+        }
+
+        /// <summary>
+        /// Retrieves the <see cref="FunctionConfiguration"/> from the context.
+        /// </summary>
+        /// <param name="testContext">The context in which the configuration is stored.</param>
+        /// <returns>The <see cref="FunctionConfiguration"/>.</returns>
+        /// <remarks>
+        /// If a <see cref="FunctionConfiguration"/> hasn't already been added to the context,
+        /// this method will create and add a new instance.
+        /// </remarks>
+        public static FunctionConfiguration GetFunctionConfiguration(SpecFlowContext testContext)
```

Comment:
Seems slightly arbitrary to call this argument `testContext` when we have `context` above.

Reply:
Resolved in 1694175.